fabric-item-api-v1/src/testmod/java/net/fabricmc/fabric/test/item/client/item/PatchedUpdatingItem.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.test.item.client.item;
+
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.api.client.item.v1.UpdateAnimationHandler;
+
+public class PatchedUpdatingItem extends UpdatingItem implements UpdateAnimationHandler {
```

Comment:
Will crash on dedicated server since `UpdateAnimationHandler` is client only. Either we change the interface to be present on all environments or mark it as implemented only on client via `@EnvionmentInterfaces`

Reply:
Making the interface available on both client and server seems like the more intuitive option, but I'll wait for any other insight or thoughts before fully committing to it.